### PR TITLE
YR/Fix build number appearing in sdk upload via pipeline/XSUP-30181

### PR DESCRIPTION
**Pull request by RosenbergYehuda**
## Related Issues
fixes:https://jira-dc.paloaltonetworks.com/browse/XSUP-30181

## Description
When uploading to the bucket, the build number is appended to the package. However, in SDK uploads, the build number should not anticipated. If a user executes the upload command within a GitLab pipeline, the pipeline number will be visible. This pull request addressed the issue by implementing validation to ensure it is a content repository before attaching the build number.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@
 * Added support to run specific hooks in **pre-commit**. Use with `demisto-sdk pre-commit <hook>`.
 * **Breaking change**: Removed the command **run-unit-tests**. Use `demisto-sdk pre-commit pytest-in-docker` instead.
 * **Breaking change**: Removed the `--unit-test` argument in **pre-commit**. To skip unit tests, run with `--no-docker` or with `skip=pytest-in-docker`,
+* Fixed an issue where the build number appeared in the pack info when using SDK-upload via pipeline.
```
Comment (GuyAfik) [inline CHANGELOG.md:16]:
```suggestion
* Fixed an issue where the build number was incorrectly shown in XSOAR marketplace when using the **upload** command on external repositories via GitLab Pipelines.

```

File: demisto_sdk/commands/common/content/tests/objects/pack_objects/pack_metadata/pack_metadata_test.py [hunk 2]
```diff
@@ -390,3 +397,69 @@ def test_load_user_metadata_bad_pack_metadata_file(repo, monkeypatch, caplog):
     pack_1_metadata.load_user_metadata("Pack1", "Pack Number 1", pack_1.path, logger)
 
     assert "Failed loading Pack Number 1 user metadata." in caplog.text
+
+
+my_instance = PackMetadata(
```
Comment (GuyAfik) [inline demisto_sdk/commands/common/content/tests/objects/pack_objects/pack_metadata/pack_metadata_test.py:402]: define it inside the test
do you need to send all of the values?

**Review by GuyAfik — COMMENTED** [review]
@RosenbergYehuda looks good, see my comments.
Also, please run `pre-commit run -a` on your PR.

File: CHANGELOG.md (hunk 1 above)
Comment (RosenbergYehuda) [reply CHANGELOG.md:16]: fixed

File: demisto_sdk/commands/common/content/tests/objects/pack_objects/pack_metadata/pack_metadata_test.py (hunk 2 above)
Comment (RosenbergYehuda) [reply demisto_sdk/commands/common/content/tests/objects/pack_objects/pack_metadata/pack_metadata_test.py:402]: Mooved inside.
It is not necessary for me to have the values, but most of them don't have a default value, and for those that do, it's a "pydentic" field that if I don't use it, I get a tester error.

Comment (RosenbergYehuda) [reply demisto_sdk/commands/common/content/tests/objects/pack_objects/pack_metadata/pack_metadata_test.py]: fixed.

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,7 +14,7 @@
 * **Breaking change**: Removed the command **run-unit-tests**. Use `demisto-sdk pre-commit pytest-in-docker` instead.
 * **Breaking change**: Removed the `--unit-test` argument in **pre-commit**. To skip unit tests, run with `--no-docker` or with `skip=pytest-in-docker`,
 * Fixed an issue where SDK commands were failing to execute correctly when supplied a path outside the Content repository.
-
+* Fixed an issue where the build number was incorrectly shown in XSOAR marketplace when using the **upload** command on external repositories via GitLab Pipelines.
 ## 1.24.0
```
Comment (GuyAfik) [inline CHANGELOG.md:18]:
```suggestion
* Fixed an issue where the build number was incorrectly shown in XSOAR marketplace when using the **upload** command on external repositories via GitLab Pipelines.

## 1.24.0
```

**Review by GuyAfik — APPROVED** [review]
🥇

File: demisto_sdk/commands/common/content/tests/objects/pack_objects/pack_metadata/pack_metadata_test.py [hunk 4]
```diff
@@ -390,3 +397,51 @@ def test_load_user_metadata_bad_pack_metadata_file(repo, monkeypatch, caplog):
...
+    searchRank=0,
+    excludedDependencies=[],
+    videos=[],
+    modules=[],
+) # type: ignore
+    mocker.patch(
+        "demisto_sdk.commands.content_graph.objects.pack_metadata.is_external_repository",
+        return_value=is_external,
+    )
+    os.environ["CI_PIPELINE_ID"] = "123"
```
Comment (JasBeilin) [inline demisto_sdk/commands/common/content/tests/objects/pack_objects/pack_metadata/pack_metadata_test.py:441]: Please do this in a setup and teardown structure as you can influence other tests.

**Review by JasBeilin — COMMENTED** [review]
Looks good, please see my comment about the test.

File: demisto_sdk/commands/common/content/tests/objects/pack_objects/pack_metadata/pack_metadata_test.py (hunk 4 above)
Comment (GuyAfik) [reply demisto_sdk/commands/common/content/tests/objects/pack_objects/pack_metadata/pack_metadata_test.py:441]: better to use `monkeypatch` here instead, @RosenbergYehuda you can see examples for it all around the repo

Comment (RosenbergYehuda) [reply demisto_sdk/commands/common/content/tests/objects/pack_objects/pack_metadata/pack_metadata_test.py:441]: Fixed.

**Review by JasBeilin — APPROVED** [review]

**Comment by GuyAfik** [comment]
@RosenbergYehuda please update changelog location to Unreleased

**Pull request merged**